tempesta_fw/sock_srv.c

```diff
@@ -197,6 +203,10 @@ tfw_sock_srv_connect_try_later(TfwSrvConnection *srv_conn)
 	 */
 	static const unsigned long timeouts[] = { 1, 10, 100, 250, 500, 1000 };
 
+	/* Don't rearm reconnection timer if we're about to shutdown. */
+	if (unlikely(!ss_active()))
```

Comment:
The problem is with this timer only, not all timers in the system. The problem is that we should not initiate new connection attempts during shoutdown process, because we need to traverse all the existing connections and close all of them. If the connection traversing and closing processes races with a new connection establishing, then we can leave the connection open which leads to a crash. Meantime it's OK to delete other timers at appropriate module stopping.